File: app/domain/icon.py
from uuid import uuid4


class Icon:
    
    uuid: str
    name: str
    iconPackUuid: str
    file: str
    tags: list[str]
    
    def __init__(self, name: str, file: str,  tags: list[str], uuid=None):
        self.tags = tags
        self.file = file
        self.name = name
        if uuid is None:
            self.uuid = str(uuid4())
        else:
            self.uuid = uuid
            
    def to_dict(self):
        return {
            "uuid": self.uuid,
            "name": self.name,
            "file": self.file,
            "tags": self.tags
        }
    
    @classmethod
    def from_dict(cls, data):
        return cls(
            name=data["name"],
            file=data["file"],
            tags=data["tags"],
            uuid=data["uuid"]
        )
        

File: kahldeck-dashboard/src/domain/Icon.ts
import { v4 as uuidv4 } from 'uuid';

export class Icon {
    readonly uuid: string;
    name: string;
    iconPackUuid: string;
    file: string;
    tags: string[];

    constructor(name: string, file: string, tags: string[], uuid?: string) {
        this.name = name;
        this.file = file;
        this.tags = tags;
        this.uuid = uuid || uuidv4();
        this.iconPackUuid = ''; // Inicializado como cadena vacía, se puede actualizar más tarde si es necesario
    }

    toJSON(): Record<string, any> {
        return {
            uuid: this.uuid,
            name: this.name,
            file: this.file,
            tags: this.tags
        };
    }

    static fromJSON(data: Record<string, any>): Icon {
        return new Icon(
            data.name,
            data.file,
            data.tags,
            data.uuid
        );
    }
}